distribution/docker/README.md

```diff
@@ -0,0 +1,131 @@
+the [DockerBase] enum.
+
+   * Default - this is what most people use, and is based on Ubuntu
+   * UBI - the same as the default image, but based upon [RedHat's UBI
+     images][ubi], specifically their minimal flavour.
+   * Iron Bank - this is the US Department of Defence's repository of digitally
+     signed, binary container images including both Free and Open-Source
+     software (FOSS) and Commercial off-the-shelf (COTS). In practice, this is
+     another UBI build, this time on the regular UBI image, with extra
+     hardening. See below for more details.
+   * Cloud - this is mostly the same as the default image, with some notable differences:
+      * `filebeat` and `metricbeat` are included
+      * `wget` is included
+      * The `ENTRYPOINT` is just `/bin/tini`, and the `CMD` is
+        `/app/elasticsearc.sh`. In normal use this file would be bind-mounted
+        in, but the image ships a stub version of this file so that the image
+        can still be tested.
+   * Cloud ESS - this directly extends the Cloud image, and adds all ES plugins
+     that the ES build generates in an archive directory. It also sets an
+     environment variable that points at this directory. This allows plugins to
+     be installed from the archive instead of the internet, speeding up
+     deployment times.
+
+The long-term goal is for both Cloud images to be retired in favour of the
+default image.
+
+
+## Build strategy
+
+For all image flavours, the ES build implements a pipeline:
+
+   1. Construct a Docker build context
+   2. Transform the build context so that it is possible to build it locally
+   3. Build the Docker image locally
+
+Some images use (1) as the releasable artifact, some use (3).
+
+**NOTE:** "Pipeline" isn't actually the correct term - in reality, each Gradle
+task depends on the one before it. Notably, it is the transform tasks that
+depend on a locally build `.tar.gz` Elasticsearch archive.
+
+
+## Releasing on Docker Hub
+
+Elasticsearch is an [official image on Docker
+Hub](https://hub.docker.com/_/elasticsearch). On release day, we build the ES
+Docker image and upload it to [Elastic's Docker
+registry](https://www.docker.elastic.co/). Separately, we submit a build context
+to Docker. They then build the image, and upload it to Docker Hub.
+Unfortunately, this is an asynchronous process, and we don't hear back if
+there's a failure, so even when everything works, there's a lag between
+releasing a new version of Elasticsearch, and the image being available on
+Docker Hub.
+
+Being an official image puts additional constraints on how the Elasticsearch
+image is built.
+
+   * It must extend another official image
+   * It must fetch any required artifacts - they cannot be supplied in the build
+     context.
+   * It must be platform-independent i.e. it can build on ARM and x64
+
+
+Step (2) in the build strategy above replaces the `curl` command in the
```

Comment:
nit: We're referring to step "2" but the above list is a bulleted one. Should we make it an ordered list?

Reply:
Actually it's referring to a list higher up the page. I've reworked it anyway and added a link.